File: go/queue/queue_test.go
package queue

import (
	"testing"

	"gotest.tools/v3/assert"
)

func TestQueue(t *testing.T) {
	q := New()
	q.Print()
	for i := 0; i < 10; i++ {
		q.Enqueue(i)
	}
	q.Print()
}

func TestQueue2(t *testing.T) {
	q := New()

	assert.Assert(t, q.IsEmpty(), "queue should be empty")
}

func TestQueue3(t *testing.T) {
	q := New()
	for i := 0; i < 4; i++ {
		q.Enqueue(i)
	}
	assert.Assert(t, q.Size() == 4, "queue size should be 10")
	assert.Assert(t, q.Dequeue() == 0, "queue front should be 0")
	assert.Equal(t, q.Dequeue(), 1)
	assert.Equal(t, q.Dequeue(), 2)
	assert.Equal(t, q.Dequeue(), 3)
	assert.Equal(t, q.IsEmpty(), true)

}

File: typescript/queue/queue.test.ts
import { describe, expect, it } from 'bun:test'
import { Queue } from './queue'

describe('queue', () => {

    it('should be empty', () => {
        const queue = new Queue()
        expect(queue.isEmpty()).toBe(true)
    })

    it('should not be empty', () => {
        const queue = new Queue()
        queue.enqueue(1)
        expect(queue.isEmpty()).toBe(false)
    })

    it('should add nodes', () => {
        const queue = new Queue()
        queue.enqueue(1)
        queue.enqueue(2)
        queue.enqueue(3)
        expect(queue.size()).toBe(3)
    })

    it('should enqueue nodes', () => {
        const queue = new Queue()
        queue.enqueue(1)
        queue.enqueue(2)
        queue.enqueue(3)
        expect(queue.dequeue()).toBe(1);
        expect(queue.dequeue()).toBe(2);
        expect(queue.dequeue()).toBe(3);
        expect(queue.isEmpty()).toBe(true)
    })
})